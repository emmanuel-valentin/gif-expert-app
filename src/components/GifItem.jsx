import PropTypes from 'prop-types'

const GifItem = ({ title, image }) => {
  return (
    <div className='card'>
      <p>{ title }</p>
      <img src={ image.url } alt={title} />
    </div>
  );
}

GifItem.prototype = {
  title: PropTypes.string.isRequired,
  image: PropTypes.object.isRequired
}
export default GifItem;