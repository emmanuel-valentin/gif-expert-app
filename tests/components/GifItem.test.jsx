import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('Pruebas en el componente GifItem', () => {
  const title = "Hola mundo";
  const image = 'https://www.holamundo.com/';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={image} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={image} />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(image);
    expect(alt).toBe(title);
  });

  test('Debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem title={title} url={image} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});