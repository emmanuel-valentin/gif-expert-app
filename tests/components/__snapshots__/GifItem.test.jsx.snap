// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Hola mundo
    </p>
    <img
      alt="Hola mundo"
      src="https://www.holamundo.com/"
    />
  </div>
</div>
`;
